src/HotChocolate/Filters/src/Types.Filters/FilterOperation.cs

```diff
@@ -11,20 +12,32 @@ public class FilterOperation
 
         public FilterOperation(
             Type type,
+            FilterKind filterKind,
             FilterOperationKind kind,
             PropertyInfo property)
         {
             Type = type ?? throw new ArgumentNullException(nameof(type));
+            FilterKind = filterKind;
             Kind = kind;
             Property = property ?? throw new ArgumentNullException(nameof(property));
+
+            if (Property.DeclaringType != null)
+            {
+                IsNullable = new NullableHelper(
+                    Property.DeclaringType).GetPropertyInfo(Property).IsNullable;
+            }
         }
 
         public Type Type { get; }
 
+        public FilterKind FilterKind { get; }
```

Comment:
@michaelstaib  Should we leave FilterKind and FilerOperationKind enums? or should we convert them to `object`. There might be some extra boxing but it would enable people to create their own identifieres. 
Have a look at how man operation on GEO JSON ire possible :
https://www.npgsql.org/efcore/mapping/nts.html

Reply:
for this pr we will leave it. I want to get this in. After that we can discuss what we will do with it.

Reply:
can you open an issue for this